### PR TITLE
fix #14604 CompareValidator compareAttribute does not work if `compar…

### LINKED ISSUE #14604: CompareAttribute does not work when you change input ID to compare against
### What steps will reproduce the problem?
Create simple form model with attributes;
newPassword
newPasswordRepeat

make Compare attribute on top of them, so that newPasswordRepeat checks if it is the same as newPassword

on view change newPassword ID
### What is the expected result?
newPasswordRepeat should check against forms newPassword
### What do you get instead?
since it assumes that no-one changes input ID s it will never be true

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      |  5.6.30
| Operating system | Ubuntu 16.10

### PR DESCRIPTION
**Pull request by mikk150**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14604

### CODE REVIEW

File: framework/assets/yii.validation.js [hunk 1]
```diff
@@ -273,7 +273,12 @@ yii.validation = (function ($) {
             if (options.compareAttribute === undefined) {
                 compareValue = options.compareValue;
             } else {
-                compareValue = $('#' + options.compareAttribute).val();
+                var attributes = $form.data('yiiActiveForm').attributes
+                for (var i = attributes.length - 1; i >= 0; i--) {
+                    if (attributes[i].id == options.compareAttribute) {
```
Comment (samdark) [inline framework/assets/yii.validation.js]: `===` could be used.

**Review by samdark — CHANGES_REQUESTED** [review]
Overall makes sense.

File: framework/assets/yii.validation.js (hunk 1 above)
Comment (mikk150) [inline framework/assets/yii.validation.js]: yep, that is true

File: framework/assets/yii.validation.js [hunk 2]
```diff
@@ -273,7 +273,12 @@ yii.validation = (function ($) {
             if (options.compareAttribute === undefined) {
                 compareValue = options.compareValue;
             } else {
-                compareValue = $('#' + options.compareAttribute).val();
+                var attributes = $form.data('yiiActiveForm').attributes
+                for (var i = attributes.length - 1; i >= 0; i--) {
```
Comment (mikk150) [inline framework/assets/yii.validation.js]: 1st speed (do not need to count every time, just simple substraction)
2nd compatible with current version, which takes first element's value

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**